### PR TITLE
[Feature]: Reformat resize config

### PR DESCRIPTION
**Pull request by YuanLiuuuuuu**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

**Before PR**:

- [ ] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmcv/transforms/processing.py [hunk 1]
```diff
@@ -1336,14 +1320,13 @@ def _random_sample_ratio(
         scale = int(scale[0] * ratio), int(scale[1] * ratio)
         return scale
 
-    def _random_scale(self, results: dict) -> None:
+    @cacheable_method
+    def _random_scale(self) -> None:
```
Comment (mzr1996) [inline mmcv/transforms/processing.py:1324]: return Tuple

File: mmcv/transforms/processing.py [hunk 2]
```diff
@@ -1336,14 +1320,13 @@ def _random_sample_ratio(
...
         """Private function to randomly sample an scale according to the type
-        of `scale`.
+        of ``scale``.
 
-        Args:
-            results (dict): Result dict from :obj:`dataset`.
         Returns:
-            dict: One new key 'scale`is added into ``results``,
+            dict: One new key ``scale`` is added into ``results``,
```
Comment (mzr1996) [inline mmcv/transforms/processing.py:1329]: The return type is not dict now.

**Review by mzr1996 — CHANGES_REQUESTED** [review]

File: mmcv/transforms/processing.py (hunk 1 above)
Comment (YuanLiuuuuuu) [reply mmcv/transforms/processing.py:1324]: I think it's tuple, instead of Tuple.

**Review by mzr1996 — APPROVED** [review]
Looks good

File: mmcv/transforms/processing.py [hunk 3]
```diff
@@ -1251,58 +1250,43 @@ class RandomResize(BaseTransform):
...
-            outside the border of the image. In some dataset like MOT17, the
-            gt bboxes are allowed to cross the border of images. Therefore,
-            we don't need to clip the gt bboxes in these cases.
-            Defaults to True.
-        backend (str): Image resize backend, choices are 'cv2' and 'pillow'.
-            These two backends generates slightly different results. Defaults
-            to 'cv2'.
-        interpolation (str): How to interpolate the original image when
-            resizing. Defaults to 'bilinear'.
+        resize_cfg (dict): Config to initialize a ``Resize`` object.
```
Comment (MeowZheng) [inline mmcv/transforms/processing.py:1257]: might present default value
a ``Resize`` object -> a ``Resize`` transform?

**Comment by Harold-lkk** [comment]
LGTM

File: mmcv/transforms/processing.py [hunk 4]
```diff
@@ -1217,18 +1218,16 @@ def __repr__(self) -> str:
...
-      sampled from [`scale[0][0]`, `scale[1][0]`] uniformally and the second
-      value of the target scale is sampled from [`scale[0][1]`, `scale[1][1]`]
-      uniformally.
-    - if `scale` is a tuple, the first and second values of the target scale
-      is equal to the first and second values of `scale` multiplied by a value
-      sampled from [`ratio_range[0]`, `ratio_range[1]`] uniformally.
+    - if ``scale`` is a list of tuple, the first value of the target scale is
+      sampled from [``scale[0][0]``, ``scale[1][0]``] uniformally and the
+      second value of the target scale is sampled from
+      [``scale[0][1]``, ``scale[1][1]``] uniformally.
```
Comment (MeowZheng) [inline mmcv/transforms/processing.py:1227]: might mention 'follow xx convention,scale[0][0] is for height? and  scale[1][0] for width?'

**Review by MeowZheng — APPROVED** [review]

**Review by Harold-lkk — APPROVED** [review]

**Pull request merged**